hack/e2e.sh

```diff
@@ -89,18 +99,30 @@ PE_URL=$(oc get route cincinnati-policy-engine -o jsonpath='{.spec.host}')
 export GRAPH_URL="http://${PE_URL}/api/upgrades_info/v1/graph"
 
 # Wait for route to become available
-ATTEMPTS=10
 DELAY=10
-
-while [ $ATTEMPTS -ge 0 ]; do
+for i in $(seq 1 10); do
   CODE=$(curl -s -o /dev/null -w "%{http_code}" --header 'Accept:application/json' "${GRAPH_URL}?channel=a")
   if [ "${CODE}" == "200" ]; then
     break
-  else
-    sleep ${DELAY}
-    ATTEMPTS=$((ATTEMPTS-1))
   fi
+  sleep ${DELAY}
+done
+
+# Wait for cincinnati metrics to be recorded
+# Find out the token Prometheus uses from its serviceaccount secrets
+# and use it to query for GB build info
+export PROM_ENDPOINT=$(oc -n openshift-monitoring get route thanos-querier -o jsonpath="{.spec.host}")
+export PROM_TOKEN=$(oc -n openshift-monitoring get secret \
+  $(oc -n openshift-monitoring get serviceaccount prometheus-k8s \
+    -o jsonpath='{range .secrets[*]}{.name}{"\n"}{end}' | grep prometheus-k8s-token) \
+  -o go-template='{{.data.token | base64decode}}')
+
+DELAY=30
+for i in $(seq 1 10); do
+  curl -kLs -H "Authorization: Bearer ${PROM_TOKEN}" "https://${PROM_ENDPOINT}/api/v1/query?query=cincinnati_gb_build_info" | grep "metric" && break
```

Comment:
Since `set -euo pipefail` is set above the test will fail with this call. Is this intended?

Reply:
Oh, good catch. Pushed a fix for this